src/node/desktop/src/main/application.ts

```diff
@@ -206,6 +206,12 @@ export class Application implements AppState {
         }
         return;
       }
+      else {
+        // for non-Rproj files, we want to set the initial working directory, before opening the file
```

Comment:
```suggestion
        // for non-Rproj files, we want to set the initial working directory, before opening the file
        // if a session does not already exist (otherwise, just open the file)
```